src/triggered/CommunicationMapping.ts

```diff
@@ -0,0 +1,125 @@
+import { LitElement } from 'lit-element';
+
+import { getDataSink, newWizardEvent } from '../foundation.js';
+
+import { commMappingWizard } from './wizards.js';
+
+export type ControlBlockType =
+  | 'ReportControl'
+  | 'LogControl'
+  | 'GSEControl'
+  | 'SampledValuesControl';
+
+type ControlBlockConnectionType = 'IEDName' | 'ClientLN';
+
+const cbConnection: Record<ControlBlockType, ControlBlockConnectionType> = {
+  ReportControl: 'ClientLN',
+  LogControl: 'ClientLN',
+  GSEControl: 'IEDName',
+  SampledValuesControl: 'IEDName',
+};
+
+export interface Connection {
+  source: Element;
+  data?: Element;
+  sink: Element;
+}
+
+export function getConnectionIndexOf(
+  list: Connection[],
+  match: Connection
+): number {
+  for (let i = 0; list.length; i++)
+    if (
+      list[i].source.closest('IED')?.getAttribute('name') ===
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/openscd/open-scd/pullRequest?prid=7203429)
```suggestion
      list[i].source.closest("IED")?.getAttribute('name') ===
```